src/commonMain/kotlin/org/kbson/BsonArray.kt

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright 2008-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.kbson
+
+/** A type-safe representation of the BSON array type. */
+class BsonArray(initial: List<BsonValue> = emptyList()) : BsonValue(), List<BsonValue> {
```

Comment:
The Java Bson array is a mutable array. This implementation only implements the `List` interface, which e.g. doesn't include `add`. I assume this should have been `MutableList`?

Reply:
Good catch - thanks